usd/src/usd_parser/USDLinks.cc

```diff
@@ -117,22 +146,419 @@ void GetInertial(
+  {
+    subMesh.AddIndex(indexes[i]);
+  }
+
+  for (auto & textCoord : textCoords)
+  {
+    subMesh.AddTexCoord(textCoord[0], (1 - textCoord[1]));
+  }
+
+  for (auto & point : points)
+  {
+    ignition::math::Vector3d v =
+      ignition::math::Vector3d(point[0], point[1], point[2]) * metersPerUnit;
+    subMesh.AddVertex(v);
+  }
+
+  for (auto & normal : normals)
+  {
+    subMesh.AddNormal(normal[0], normal[1], normal[2]);
+  }
+
+  sdf::Mesh meshGeom;
+  _geom.SetType(sdf::GeometryType::MESH);
+
+  ignition::math::Pose3d pose;
+  ignition::math::Vector3d scale(1, 1, 1);
+  std::string linkName = _link->Name();
+  size_t nSlash = std::count(linkName.begin(), linkName.end(), '/');
+  if (nSlash == 1)
+  {
+    GetTransform(_prim, _usdData, pose, scale, "/");
+  }
+  else
+  {
+    GetTransform(_prim, _usdData, pose, scale, _link->Name());
+  }
+  meshGeom.SetScale(scale * _scale);
+
+  std::string primName = pxr::TfStringify(_prim.GetPath());
+
+  std::string directoryMesh = directoryFromUSDPath(primName) + primName;
+
+  meshGeom.SetFilePath(
+    ignition::common::joinPaths(
+      directoryMesh, ignition::common::basename(directoryMesh)) + ".dae");
+
+  meshGeom.SetUri(meshGeom.FilePath());
+
+  int numSubMeshes = ParseMeshSubGeom(
+    _prim, _link, subMesh, meshGeom, _scale, _usdData);
+
+  _geom.SetMeshShape(meshGeom);
+
+  if (numSubMeshes == 0)
+  {
+    std::string nameMaterial = ParseMaterialName(_prim);
+    auto it = _usdData.Materials().find(nameMaterial);
+    if (it != _usdData.Materials().end())
+    {
+      _vis.SetMaterial(it->second);
+      // link->visual_array_material.push_back(it->second);
+      std::shared_ptr<ignition::common::Material> matCommon =
+        std::make_shared<ignition::common::Material>();
+      convert(it->second, *matCommon.get());
+      mesh.AddMaterial(matCommon);
+      subMesh.SetMaterialIndex(mesh.MaterialCount() - 1);
+    }
+
+    std::string primNameStr = _prim.GetPath().GetName();
+    _vis.SetName(primNameStr + "_visual");
+    _vis.SetRawPose(pose);
+    _vis.SetGeom(_geom);
+
+    _link->AddVisual(_vis);
+
+    // link->visual_array_material_name.push_back(nameMaterial);
+    mesh.AddSubMesh(subMesh);
+
+    if (ignition::common::createDirectories(directoryMesh))
+    {
+      directoryMesh = ignition::common::joinPaths(
+        directoryMesh, ignition::common::basename(directoryMesh));
+      // Export with extension
+      ignition::common::ColladaExporter exporter;
+      exporter.Export(&mesh, directoryMesh, false);
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+/// \brief Parse USD cube
+/// \param[in] _prim Path of the prim
+/// \param[in] _geom sdf geom
+/// \param[in] _scale scale mesh
+/// \param[in] _metersPerUnit meter per unit of the stage
+void ParseCube(const pxr::UsdPrim &_prim,
+  sdf::Geometry &_geom,
+  ignition::math::Vector3d &_scale,
+  const double _metersPerUnit)
+{
+  double size;
+  auto variant_cylinder = pxr::UsdGeomCube(_prim);
```

Comment:
Should this variable be renamed to `variant_cube`, since this method parses a USD cube?